runner.py

```diff
@@ -135,17 +136,18 @@ def stop(*args):
     routes = await proxy.get_all_routes()
     return urls, routes
 
-
-def single_run_http(url, delay, size, msgs):
-    """Time a single http request"""
-    tic = time.time()
+words = []
+def my_setup(url):
+    global words
     with urlopen(url) as f:
-        f.read()
-    toc = time.time()
-    return toc - tic
+        words = f.readlines()
 
+def time_http():
+    for word in words:
+        word.http()
+time_http.setup = my_setup
 
-def single_run_ws(url, delay, size, msgs):
+def time_ws(url, delay, size, msgs):
```

Comment:
Instead of replacing single_run_ws, try writing  a `time_ws` benchmark that *calls* single_run_ws with a particular collection of arguments.